pkg/api/v1/status/atlascluster.go

```diff
@@ -0,0 +1,55 @@
+package status
+
+import "go.mongodb.org/atlas/mongodbatlas"
+
+// AtlasClusterStatus defines the observed state of AtlasCluster.
+type AtlasClusterStatus struct {
+	Common            `json:",inline"`
+	StateName         string             `json:"stateName,omitempty"`
+	MongoDBVersion    string             `json:"mongoDBVersion,omitempty"`
+	ConnectionStrings *ConnectionStrings `json:"connectionStrings"`
+	MongoURIUpdated   string             `json:"mongoURIUpdated,omitempty"`
+}
+
+// ConnectionStrings is a copy of mongodbatlas.ConnectionStrings for deepcopy compatibility purposes.
+type ConnectionStrings struct {
+	Standard          string            `json:"standard,omitempty"`
+	StandardSrv       string            `json:"standardSrv,omitempty"`
+	AwsPrivateLink    map[string]string `json:"awsPrivateLink,omitempty"`
+	AwsPrivateLinkSrv map[string]string `json:"awsPrivateLinkSrv,omitempty"`
+	Private           string            `json:"private,omitempty"`
+	PrivateSrv        string            `json:"privateSrv,omitempty"`
+}
+
+// check that the two types are compatible.
+var _ = ConnectionStrings(mongodbatlas.ConnectionStrings{})
```

Comment:
nice!
BTW can we use the same approach when creating a `mongodbatlas` cluster from our spec instead of serialization/deserialization?

Reply:
We can't, since their fields are not _exactly_ the same type because of nested structs (even though they are compatible)

Reply:
Added these checks wherever possible